package/MDAnalysis/topology/XYZParser.py

```diff
@@ -92,7 +93,7 @@ def parse(self, **kwargs):
 
         # Guessing time
         atomtypes = guessers.guess_types(names)
-        masses = guessers.guess_masses(atomtypes)
+        masses = guessers.guess_masses(names)
```

Comment:
Here I assume that `names` contain element symbols (XYZ file format). As discusses in #2420 this might not always be the case. Should I make this more flexible?

Reply:
I think (hope) guess_masses is smart enough to skip a bad value and print a warning?